### PR TITLE
add section for enforcing Multi-factor Authentication

### PR DESCRIPTION
**Pull request by mutec**
/cc @TimWolla

### CODE REVIEW

**Comment by TimWolla** [comment]
Apparently I did not receive a notification for this (is this due to this being a draft?), so I missed it and just saw it by accident. I just approved the actions.

File: docs/migration/wsc53/session.md [hunk 1]
```diff
@@ -142,6 +142,18 @@ Please check [the version history](https://github.com/WoltLab/WCF/commits/master
...
 !!! warning "Multi-factor authentication is security sensitive. Make sure to carefully read the remarks in `IMultifactorMethod` for possible issues. Also make sure to carefully test your implementation against all sorts of incorrect input and consider attack vectors such as race conditions. It is strongly recommended to generously check the current state by leveraging assertions and exceptions."
 
+#### Enforcing Multi-factor Authentication
+
+To enforce Multi-factor Authentication within your controller you need to:
+
+1. Use the `wcf\system\user\multifactor\TMultifactorRequirementEnforcer` trait.
+2. Call: `$this->enforceMultifactorAuthentication();`
+
+`enforceMultifactorAuthentication()` will check if the user is in a group that requires multi-factor authentication and if the user does not have multi-factor authentication enabled.
```
Comment (TimWolla) [inline docs/migration/wsc53/session.md]:
```suggestion
`enforceMultifactorAuthentication()` will check if the user is in a group that requires multi-factor authentication, but does not yet have multi-factor authentication enabled.
```

File: docs/migration/wsc53/session.md [hunk 2]
```diff
@@ -142,6 +142,18 @@ Please check [the version history](https://github.com/WoltLab/WCF/commits/master
...
+If they did, the request proceeds as usual.
+Otherwise, a `wcf\system\exception\NamedUserException` is thrown.
```
Comment (TimWolla) [inline docs/migration/wsc53/session.md]:
```suggestion
Otherwise, a `NamedUserException` is thrown.
```

**Review by TimWolla — APPROVED** [review]
Overall LGTM. Is there a reason this is a draft?

**Comment by mutec** [comment]
> Apparently I did not receive a notification for this (is this due to this being a draft?), so I missed it and just saw it by accident. I just approved the actions.

Wouldn't make sense since I explicitly tagged you. But not everything is meant to make sense in this world. ^^

> Overall LGTM. Is there a reason this is a draft?

I thought it would make sense if Marcel or Alexander might want to have a look, too. But yes, it's kinda stupid since it's a very small change.

**Comment by TimWolla** [comment]
> I thought it would make sense if Marcel or Alexander might want to have a look, too.

Draft PRs imply that the author might want to make additional changes before review / approval. If review from multiple persons is desired we'll just request the appropriate review.

**Review by TimWolla — APPROVED** [review]

**Comment by TimWolla** [comment]
Thanks!

**Pull request merged**